test_test.go

```diff
@@ -71,6 +73,66 @@ func TestPost(t *testing.T) {
 	}
 }
 
+func TestPut(t *testing.T) {
+	test := NewTest("unit-test")
+
+	sample := SampleObject{}
+	test = test.Put(api.URL, "/tests", nil).ParseResponseBody(&sample)
+	if sample.Name != "unit-test" {
+		t.Errorf("name response was %s, expected unit-test", sample.Name)
+	}
+	if !sample.Success {
+		t.Error("expected response success to be true")
+	}
+}
+
+func TestDelete(t *testing.T) {
+	test := NewTest("unit-test")
+
+	test = test.Delete(api.URL, "/tests")
+	if test.Status != http.StatusNoContent {
+		t.Errorf("expected status 204 No Content, instead was %d", test.Status)
+	}
+}
+
+func TestMalformedUrl(t *testing.T) {
+	test := NewTest("unit-test")
+
+	test = test.Get(api.URL, "bad-url%@%(*%)///\\####")
+	if test.Error == nil {
+		t.Errorf("expected an error, but did not get one")
+	}
+}
+
+func TestBadRequestBody(t *testing.T) {
+	test := NewTest("unit-test")
+
+	// pass a clearly bogus request body to force a json parse error
+	test = test.Post(api.URL, "/tests", make(chan int))
```

Comment:
i think this is a much better approach than passing a channel in to make the parser fail.